vpr/src/route/router_lookahead_map.cpp

```diff
@@ -444,66 +454,31 @@ std::pair<float, float> MapLookahead::get_expected_delay_and_cong(RRNodeId from_
-            //router. So just return an arbitrary value here rather than error.
-
-            //We choose to return the largest (non-infinite) value possible, but scaled
-            //down by a large factor to maintain some dynaimc range in case this value ends
-            //up being processed (e.g. by the timing analyzer).
-            //
-            //The cost estimate should still be *extremely* large compared to a typical delay, and
-            //so should ensure that the router de-prioritizes exploring this path, but does not
-            //forbid the router from trying.
-            expected_delay_cost = std::numeric_limits<float>::max() / 1e12;
-            expected_cong_cost = std::numeric_limits<float>::max() / 1e12;
-        } else {
-            //From the current SOURCE/OPIN we look-up the wiretypes which are reachable
-            //and then add the estimates from those wire types for the distance of interest.
-            //If there are multiple options we use the minimum value.
-            for (const auto& kv : this->src_opin_delays[from_layer_num][tile_index][from_ptc]) {
-                const util::t_reachable_wire_inf& reachable_wire_inf = kv.second;
-
-                Cost_Entry wire_cost_entry;
-                if (reachable_wire_inf.wire_rr_type == SINK) {
-                    //Some pins maybe reachable via a direct (OPIN -> IPIN) connection.
-                    //In the lookahead, we treat such connections as 'special' wire types
-                    //with no delay/congestion cost
-                    wire_cost_entry.delay = 0;
-                    wire_cost_entry.congestion = 0;
-                } else {
-                    //For an actual accessible wire, we query the wire look-up to get it's
-                    //delay and congestion cost estimates
-                    wire_cost_entry = get_wire_cost_entry(reachable_wire_inf.wire_rr_type,
-                                                          reachable_wire_inf.wire_seg_index,
-                                                          from_layer_num,
-                                                          delta_x,
-                                                          delta_y);
-                }
-
-                float this_delay_cost = (params.criticality) * (reachable_wire_inf.delay + wire_cost_entry.delay);
-                float this_cong_cost = (1. - params.criticality) * (reachable_wire_inf.congestion + wire_cost_entry.congestion);
-
-                expected_delay_cost = std::min(expected_delay_cost, this_delay_cost);
-                expected_cong_cost = std::min(expected_cong_cost, this_cong_cost);
-            }
+        // Currently, we assume inter-layer connections are only from a block output pin to another layer. Thus, if the from and to layers are different,
+        // We use src_opin_inter_layer_delays.
+        if (from_layer_num == to_layer_num) {
```

Comment:
Is the if statement for speed?  Otherwise a degenerate look-up with both the from to layer numbers being the same would work.  If there's no speed advantage, I suggest combining the code (always go through the more general data structure) to minimize the number of code paths.

Reply:
That is for speed. In the updated router lookahead (6d array) a general data structure is used.